docs/articles/concepts/joins.md

```diff
@@ -1,3 +1,775 @@
+        {
+            "user_id": 123,
+            "at": datetime(2023, 12, 14, 0, 15, tzinfo=timezone.utc),
+            "email": "bee@bytewax.io",
+        },
+        {
+            "user_id": 123,
+            "at": datetime(2023, 12, 14, 0, 30, tzinfo=timezone.utc),
+            "email": "queen@bytewax.io",
+        },
+    ]
+)
+
+bytewax.testing.run_main(flow)
+```
+
+Notice how now we only have the latest email for the bee:
+
+```{testoutput}
+join_eg.check_join: ('123', (WindowMetadata(open_time: 2023-12-14 00:00:00 UTC, close_time: 2023-12-14 01:00:00 UTC), ({'user_id': 123, 'at': datetime.datetime(2023, 12, 14, 0, 0, tzinfo=datetime.timezone.utc), 'name': 'Bee'}, {'user_id': 123, 'at': datetime.datetime(2023, 12, 14, 0, 30, tzinfo=datetime.timezone.utc), 'email': 'queen@bytewax.io'})))
+```
+
+Now if we re-define the dataflow and use `product=True`, we can see
+all of the values for the Bee's email in that window.
+
+```python
+import bytewax.operators.window as op_w
+
+flow = Dataflow("join_eg")
+
+names = op.input("names", flow, TestingSource(names_l))
+emails = op.input("emails", flow, TestingSource(emails_l))
+
+keyed_names = op.map("key_names", names, lambda x: (str(x["user_id"]), x))
+keyed_emails = op.map("key_emails", emails, lambda x: (str(x["user_id"]), x))
+
+joined = op_w.join_window(
+    "join", clock, windower, keyed_names, keyed_emails, product=True
+)
+
+op.inspect("check_join", joined)
+
+bytewax.testing.run_main(flow)
+```
+
+Notice there are now two output values for that key.
+
+```{testoutput}
+join_eg.check_join: ('123', (WindowMetadata(open_time: 2023-12-14 00:00:00 UTC, close_time: 2023-12-14 01:00:00 UTC), ({'user_id': 123, 'at': datetime.datetime(2023, 12, 14, 0, 0, tzinfo=datetime.timezone.utc), 'name': 'Bee'}, {'user_id': 123, 'at': datetime.datetime(2023, 12, 14, 0, 15, tzinfo=datetime.timezone.utc), 'email': 'bee@bytewax.io'})))
+join_eg.check_join: ('123', (WindowMetadata(open_time: 2023-12-14 00:00:00 UTC, close_time: 2023-12-14 01:00:00 UTC), ({'user_id': 123, 'at': datetime.datetime(2023, 12, 14, 0, 0, tzinfo=datetime.timezone.utc), 'name': 'Bee'}, {'user_id': 123, 'at': datetime.datetime(2023, 12, 14, 0, 30, tzinfo=datetime.timezone.utc), 'email': 'queen@bytewax.io'})))
+```
+
+## Named Joins
+
+The two previously described join operators have **named** versions.
+[`join`](/apidocs/bytewax.operators/index#bytewax.operators.join), has
+[`join_named`](/apidocs/bytewax.operators/index#bytewax.operators.join_named)
+and
+[`join_window`](/apidocs/bytewax.operators/window/index#bytewax.operators.window.join_window),
+has
+[`join_window_named`](/apidocs/bytewax.operators/window/index#bytewax.operators.window.join_window).
+The named versions have identical parameters and join semantics, but
+in stead of emitting `tuple`s down stream, they emit `dict`s in which
+the keys are the names of the keyword arguments you use to specify the
+upstream sides. This helps out a lot when you many-sided joins and it
```

Comment:
(couldn't help but notice that this is similar to the infamous meme ["I accidently X. The whole thing"](https://knowyourmeme.com/memes/i-accidentally).
Probably you want to change the last sentence of this paragraph